app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/LayoutCalculatorTest.kt

```diff
@@ -145,13 +133,14 @@ class LectureViewDrawerTest {
         val startTime1 = 23 * 60 // 11:00pm
         val startTime2 = startTime1 + duration1 + 20 // 00:05am, next day
 
-        val lecture1 = createLecture(roomIndex = 0, date = conferenceDate, startTime = startTime1, duration = duration1)
-        val lecture2 = createLecture(roomIndex = 1, date = conferenceDate, startTime = startTime2)
-        val lectures = listOf(lecture1, lecture2)
-        conference.calculateTimeFrame(lectures) { Moment(it).minuteOfDay }
+        val lecture1 = createLecture(date = conferenceDate, startTime = startTime1, duration = duration1)
+        val lecture2 = createLecture(date = conferenceDate, startTime = startTime2)
+        val lecturesInRoom0 = listOf(lecture1)
+        val lecturesInRoom1 = listOf(lecture2)
+        val conference = Conference(firstEventStartsAt = startTime1, lastEventEndsAt = startTime2)
```

Comment:
Agree with the data class part. I think, it would be "minutes since midnight of conference day" though (lecture of day 1 take place after 24:00).
I also see your point with "testing two classes". But I would not have done it that way anyway, since calculateTimeFrame is already unit tested and the intent of the Conference class is to not manually set its member values, but use the calculation method. Now, chances are higher, that someone does not correctly set the start and end values. It took me quiet some time, to go through calculateTimeFrame step by step and make sure (as best as my brain can assure such thing) that your changes are equivalent.
Nevertheless I can live with your changes :).

Reply:
I don't see the value in not allowing tests to manually create `Conference` instances with desired values. There's nothing that forces `LayoutCalculator` to only work with lecture instances used with `Conference.calculateTimeFrame()`. So the only effect of not using the constructor is that it's more cumbersome to create `Conference` instances in tests. You could still create `Conference` instances that have "wrong" values, e.g. a conference start time after a lecture start time and/or a conference end time before a lecture end time (if different `Lecture` instances are used with `Conference.calculateTimeFrame()` and `LayoutCalculator.calculateLayoutParams()`).

Not setting the `Conference` values "correctly" is not really a problem. Either we have passing tests that check inputs that can never occur in production. Or we get failing tests because `LayoutCalculator` was changed to make use of a property of `Conference` it didn't use before. In that case failing tests would be a good thing because it reminded us to add at least one new test.

Apologies for making this hard to review. It is a change that was not strictly necessary. But I think it cleans up the tests nicely. 
A way to avoid having to go through `calculateTimeFrame()` step by step is to run the old tests and print out the `Conference` values after the call to `calculateTimeFrame()`.